Mathlib/Algebra/Polynomial/Laurent.lean

```diff
@@ -512,6 +510,80 @@ theorem isLocalization : IsLocalization (Submonoid.powers (X : R[X])) R[T;T⁻¹
       rintro rfl
       exact ⟨1, rfl⟩ }
 
+theorem mk'_mul_T (p : R[X]) (n : ℕ) :
+  IsLocalization.mk' R[T;T⁻¹] p (⟨X^n,n,rfl⟩ : Submonoid.powers (X : R[X])) * T n =
+    toLaurent p := by
+  rw [←toLaurent_X_pow, ←algebraMap_eq_toLaurent, IsLocalization.mk'_spec, algebraMap_eq_toLaurent]
+
+theorem mk'_eq (p : R[X]) (n : ℕ) : IsLocalization.mk' R[T;T⁻¹] p
+  (⟨X^n,n,rfl⟩ : Submonoid.powers (X : R[X])) = toLaurent p * T (-n) := by
+  rw [←IsUnit.mul_left_inj (isUnit_T n), mul_T_assoc, neg_add_cancel, T_zero, mul_one]
+  exact mk'_mul_T p n
+
+theorem mk'_one_X_pow (n : ℕ) : IsLocalization.mk' R[T;T⁻¹] 1
+  (⟨X^n,n,rfl⟩ : Submonoid.powers (X : R[X])) = T (-n) := by
+  rw [mk'_eq 1 n, toLaurent_one, one_mul]
+
+theorem mk'_one_X : IsLocalization.mk' R[T;T⁻¹] 1
+  (⟨X,1,pow_one X⟩ : Submonoid.powers (X : R[X])) = T (-1) := by
+  convert mk'_one_X_pow 1
+  exact (pow_one X).symm
+
+/-- Given a ring homomorphism `f : R →+* S` and a unit `x` in `S`, the induced homomorphism
+`R[T;T⁻¹] →+* S` sending `T` to `x` and `T⁻¹` to `x⁻¹`. -/
+def eval₂ : R[T;T⁻¹] →+* S :=
+  IsLocalization.lift (M := Submonoid.powers (X : R[X])) (g := Polynomial.eval₂RingHom f x) (by
+    rintro ⟨y, n, rfl⟩
+    simpa only [coe_eval₂RingHom, eval₂_X_pow] using IsUnit.pow n x.isUnit
+  )
+
+@[simp]
+theorem eval₂_toLaurent (p : R[X]) : eval₂ f x (toLaurent p) = Polynomial.eval₂ f x p := by
+  unfold eval₂
+  rw [←algebraMap_eq_toLaurent, IsLocalization.lift_eq]
+  rfl
+
+@[simp]
+theorem eval₂_T_n (n : ℕ) : eval₂ f x (T n) = x ^ n := by
+  rw [←Polynomial.toLaurent_X_pow, eval₂_toLaurent, eval₂_X_pow f x]
+
+@[simp]
+theorem eval₂_T_neg_n (n : ℕ) : eval₂ f x (T (-n)) = x⁻¹ ^ n := by
+  rw [←mk'_one_X_pow]
+  unfold eval₂
+  rw [IsLocalization.lift_mk'_spec, map_one, coe_eval₂RingHom, eval₂_X_pow, ←mul_pow,
+    Units.mul_inv, one_pow]
+
+theorem eval₂_T (n : ℤ) : eval₂ f x (T n) =
+  if (0 ≤ n) then (x ^ n.toNat) else (x⁻¹ ^ (-n).toNat) := by
```

Comment:
It should now be possible to simplify this as follows, right?
```suggestion
theorem eval₂_T (n : ℤ) : eval₂ f x (T n) = x ^ n
```
And it would subsume the two lemmas above, so maybe this one can be `@[simp]` and the other two would be more of an implementation detail?
Or maybe you can even simplify the proof?

Reply:
You're right, I don't need the if/else anymore and the proofs get a bit simpler. I still think the auxiliary lemmas above are useful though, as one still needs to split by cases on whether or not n is positive.